loopy/target/c/__init__.py

```diff
@@ -115,55 +115,55 @@ def _preamble_generator(preamble_info):
     function_defs = {
             "loopy_floor_div": r"""
             #define LOOPY_DEFINE_FLOOR_DIV(SUFFIX, TYPE) \
-                inline TYPE loopy_floor_div_##SUFFIX(TYPE a, TYPE b) \
-                { \
+                {} TYPE loopy_floor_div_##SUFFIX(TYPE a, TYPE b) \
+                {{ \
                     if ((a<0) != (b<0)) \
                         a = a - (b + (b<0) - (b>=0)); \
                     return a/b; \
-                }
+                }}
             LOOPY_CALL_WITH_INTEGER_TYPES(LOOPY_DEFINE_FLOOR_DIV)
             #undef LOOPY_DEFINE_FLOOR_DIV
-            """,
+            """.format(func_qualifier),
```

Comment:
I don't find `.format()` templates easy to read when they're supposed to contain braces.  `%s`? (Same below, repeatedly.)

Reply:
Slight problem there is our string that is to be formatted contains terms like "%x" which are not quite favorable to the "%" formatter, we would have to split the string into multiple lines and then apply `(%s) % str_arg` which would be equally ugly.